Source/Fuse.Controls.CameraView/Android/CameraImpl.java

```diff
@@ -253,6 +257,164 @@ public void setFlashMode(String flashMode) {
+                        designAdjustment = 0;
+                    }
+
+                    int amountApplied = (int)Math.round( (exposureAmount * (y/cameraHeight)) ) + designAdjustment; 
+                    amountApplied = (amountApplied > exposureAmount) ? exposureAmount : amountApplied;
+                    
+                    parameters.setExposureCompensation(parameters.getMinExposureCompensation() + amountApplied);
+                    
+                    parameters.setAutoExposureLock(false);
+                }
+
+
+                _camera.setParameters(parameters);
+
+
+
+                //check for lock
+                if (isFocusLocked == 1) {
+
+                    if (focusModes.contains(Camera.Parameters.FOCUS_MODE_MACRO)) {
+                        parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_MACRO);
+                    }
+
+                    if (parameters.isAutoExposureLockSupported()) {
+
+                        parameters.setAutoExposureLock(true);
+                    }
+
+                    if (parameters.isAutoWhiteBalanceLockSupported()) {
+
+                        parameters.setAutoWhiteBalanceLock(true);
+                    }
+
+                    _camera.setParameters(parameters);
+
+
+                    _autoFocus = false;
+                } else {
+                    _autoFocus = false;
+                }
+
+                _camera.startPreview();
+
+                _camera.autoFocus(new Camera.AutoFocusCallback() {
+                    @Override
+                    public void onAutoFocus(boolean success, Camera camera) {
+
+                        if (_camera.getParameters().getFocusMode().equals(Camera.Parameters.FOCUS_MODE_CONTINUOUS_PICTURE)) {
+
+                            android.hardware.Camera.Parameters parameters = _camera.getParameters();
+                            parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_CONTINUOUS_PICTURE);
+                            if (parameters.getMaxNumFocusAreas() > 0) {
+                                parameters.setFocusAreas(null);
+                            }
+                            _camera.setParameters(parameters);
+                            _camera.startPreview();
+                        }
+                    }
+                });
+
+            } catch(Exception e) {
+                android.util.Log.d(toString(), e.getMessage());
+            }
+        } 
+    }
+
+
+    private int clamp(int touchCoordinateInCameraReper, int focusAreaSize) {
+        int result;
+        if (Math.abs(touchCoordinateInCameraReper)+focusAreaSize/2>1000) {
+            if (touchCoordinateInCameraReper>0) {
+                result = 1000 - focusAreaSize/2;
+            } else {
+                result = -1000 + focusAreaSize/2;
+            }
+        } else {
+            result = touchCoordinateInCameraReper - focusAreaSize/2;
+        }
+        return result;
```

Comment:
You don't need result. Just return when you assign to it instead, should do the same thing.